tools/container/entrypoint.sh

```diff
@@ -0,0 +1,142 @@
+    #   command if and only if its value is 1. In that case, underscores in the key are replaced with dashes and
+    #   passed as-is to the node setup command (e.g., ICINGA_ACCEPT_COMMANDS=1 becomes --accept-commands).
+    # - The second case matches exclusively on the "global_zones" key. The value of this environment variable
+    #   is expected to be one or comma-separated list of zones. If the value is not empty, it will be split by
+    #   commas and each zone will be passed as a separate argument to the node setup command
+    #   (e.g., ICINGA_GLOBAL_ZONES=icinga,example becomes --global_zones icinga --global_zones example).
+    # - For the third match, the key is expected to be a key-value pair that should be passed to the node setup
+    #   command. In this case, key and value pairs are passed as-is to the node setup command (e.g.,
+    #   ICINGA_CN=icinga.example.com becomes --cn icinga.example.com).
+    # - For the fourth match, the trusted certificate is expected to be a PEM-encoded certificate that should be
+    #   written to a temporary file and passed to the node setup command.
+    # - Lastly, the CA certificate is likewise expected to be a PEM-encoded certificate that should be written to
+    #   the expected location at /var/lib/icinga2/certs/ca.crt.
+    #
+    # When encountering an environment variable prefixed with ICINGA_ that we don't know how to handle, we log it
+    # as an informational message and continue processing the next environment variable but it doesn't cause the
+    # script to fail.
+    for k in "${!ICINGA_@}"; do
+        # Strip the ICINGA_ prefix and convert the key to lowercase.
+        key=$(echo "${k#ICINGA_}" | awk '{print tolower($0)}')
+        # Get the value of the environment variable.
+        value="${!k}"
+
+        case "$key" in
+            "accept_commands" | "accept_config" | "disable_confd" | "master")
+                runNodeSetup=true
+                if [ "$value" = "1" ]; then
+                    nodeSetup+=("--${key//_/-}")
+                fi
+                ;;
+            "global_zones")
+                runNodeSetup=true
+                # Split the value by commas and pass each part as a separate argument.
+                IFS=',' read -ra zones <<< "$value"
+                for zone in "${zones[@]}"; do
+                    nodeSetup+=("--global_zones" "$zone")
+                done
+                ;;
+            "cn" | "endpoint" | "listen" | "parent_host" | "parent_zone" | "zone" | "ticket")
+                runNodeSetup=true
+                nodeSetup+=("--$key" "$value")
+                ;;
+            "trustedcert")
+                icinga2_log "information" "Writing trusted certificate to temporary file."
+                runNodeSetup=true
+                trustedCertFile=$(mktemp /tmp/trusted.cert)
+                echo "$value" > "$trustedCertFile"
```

Comment:
```suggestion
                printf "%s" "$value" > "$trustedCertFile"
```

Observe what happens with `value=-e` otherwise.

The bash build-in `echo` seems fine with `echo '-----BEGIN CERTIFICATE-----'` and doesn't mangle it. Going the `printf` way is just more reliable if you just want to print variable contents as-is.